airflow/models/dag.py

```diff
@@ -2150,15 +2273,22 @@ def create_dagrun(
                 "Creating DagRun needs either `run_id` or both `run_type` and `execution_date`"
             )
 
-        if run_type == DagRunType.MANUAL and data_interval is None and execution_date is not None:
-            data_interval = self.timetable.infer_data_interval(
-                run_after=timezone.coerce_datetime(execution_date),
+        logical_date = timezone.coerce_datetime(execution_date)
+        if data_interval is None and logical_date is not None:
+            warnings.warn(
+                "Calling `DAG.create_dagrun()` without an explicit data interval is deprecated",
+                DeprecationWarning,
+                stacklevel=2,
```

Comment:
```suggestion
                stacklevel=3,
```

Because of the decorator.

(Oh how I wish decorators could handle this automatically for us)